From 996294c9723e9e287bea33918acbf67334c0a3b5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?B=C5=82a=C5=BCej=20Szczygie=C5=82?= <spaz16@wp.pl>
Date: Sat, 2 May 2020 19:38:14 +0200
Subject: [PATCH] OpenGL: Explicitly link to the libGL.so

---
 src/qmplay2/CMakeLists.txt | 16 +++++++---------
 1 file changed, 7 insertions(+), 9 deletions(-)

diff --git a/src/qmplay2/CMakeLists.txt b/src/qmplay2/CMakeLists.txt
index 999dddff..89db2106 100644
--- a/src/qmplay2/CMakeLists.txt
+++ b/src/qmplay2/CMakeLists.txt
@@ -127,20 +127,18 @@ if(USE_OPENGL)
         opengl/opengl.qrc
     )
 
-    if(NOT WIN32)
-        if(Qt5Gui_OPENGL_IMPLEMENTATION STREQUAL GLESv2)
-            add_definitions(-DOPENGL_ES2)
-            list(APPEND OPENGL_LIBS GLESv2)
-        elseif(Qt5Gui_OPENGL_IMPLEMENTATION STREQUAL GLES)
-            message(SEND_ERROR "OpenGL|ES 1.0 is not supported!")
-        endif()
-    endif()
-
     if(APPLE)
         find_package(OpenGL REQUIRED)
         list(APPEND OPENGL_LIBS ${OPENGL_LIBRARIES})
     elseif(WIN32)
         list(APPEND OPENGL_LIBS opengl32)
+    elseif(Qt5Gui_OPENGL_IMPLEMENTATION STREQUAL GLESv2)
+        add_definitions(-DOPENGL_ES2)
+        list(APPEND OPENGL_LIBS GLESv2)
+    elseif(Qt5Gui_OPENGL_IMPLEMENTATION STREQUAL GLES)
+        message(SEND_ERROR "OpenGL|ES 1.0 is not supported!")
+    else()
+        list(APPEND OPENGL_LIBS GL)
     endif()
 endif()
 
