diff -up QMPlay2-19.12.19/src/modules/Visualizations/SimpleVis.cpp QMPlay2-19.12.19/src/modules/Visualizations/SimpleVis.cpp-om               
--- QMPlay2-19.12.19/src/modules/Visualizations/SimpleVis.cpp   2019-12-19 22:32:21.000000000 +0100
+++ QMPlay2-19.12.19/src/modules/Visualizations/SimpleVis.cpp-om 2020-03-26 18:56:37.775874455 +0100
@@ -20,6 +20,7 @@
 #include <Functions.hpp>
 
 #include <QPainter>
+#include <QPainterPath>
 
 #include <cmath>
 
diff -up QMPlay2-19.12.19/src/modules/Visualizations/FFTSpectrum.cpp QMPlay2-19.12.19/src/modules/Visualizations/FFTSpectrum.cpp-om
--- QMPlay2-19.12.19/src/modules/Visualizations/FFTSpectrum.cpp 2019-12-19 22:32:21.000000000 +0100
+++ QMPlay2-19.12.19/src/modules/Visualizations/FFTSpectrum.cpp-om2020-03-26 18:58:16.799018625 +0100
@@ -20,6 +20,7 @@
 #include <Functions.hpp>
 
 #include <QPainter>
+#include <QPainterPath>
 #include <qevent.h>
 
 extern "C"

diff -up QMPlay2-19.12.19/src/modules/AudioFilters/EqualizerGUI.cpp QMPlay2-19.12.19/src/modules/AudioFilters/EqualizerGUI.cpp-om
--- QMPlay2-19.12.19/src/modules/AudioFilters/EqualizerGUI.cpp  2019-12-19 22:32:21.000000000 +0100
+++ QMPlay2-19.12.19/src/modules/AudioFilters/EqualizerGUI.cpp-om2020-03-26 18:59:46.047835997 +0100
@@ -25,6 +25,7 @@
 #include <QLabel>
 #include <QSlider>
 #include <QPainter>
+#include <QPainterPath>
 #include <QCheckBox>
 #include <QScrollArea>
 #include <QGridLayout>
