diff -uraN QMPlay2-src-23.06.17/src/qmplay2/opengl/OpenGLInstance.hpp omv-QMPlay2-src-23.06.17/src/qmplay2/opengl/OpenGLInstance.hpp
--- QMPlay2-src-23.06.17/src/qmplay2/opengl/OpenGLInstance.hpp	2023-05-24 23:39:35.000000000 +0200
+++ omv-QMPlay2-src-23.06.17/src/qmplay2/opengl/OpenGLInstance.hpp	2023-07-21 20:35:42.026000110 +0200
@@ -22,7 +22,7 @@
 
 #include <QOpenGLContext>
 
-#if defined OPENGL_ES2 && !defined APIENTRY
+#ifndef APIENTRY
     #define APIENTRY
 #endif
 
