diff --git a/src/qmplay2/CMakeLists.txt b/src/qmplay2/CMakeLists.txt
index 1b5b6eea..df3ec776 100644
--- a/src/qmplay2/CMakeLists.txt
+++ b/src/qmplay2/CMakeLists.txt
@@ -126,6 +126,7 @@ if(USE_OPENGL)
 
     if(BUILD_WITH_QT6)
         list(APPEND OPENGL_LIBS Qt6::OpenGL Qt6::OpenGLWidgets)
+        get_target_property(QtGui_Enabled_Features Qt6::Gui QT_ENABLED_PUBLIC_FEATURES)
     endif()
 
     if(APPLE)
@@ -133,7 +134,7 @@ if(USE_OPENGL)
         list(APPEND OPENGL_LIBS ${OPENGL_LIBRARIES})
     elseif(WIN32)
         list(APPEND OPENGL_LIBS opengl32)
-    elseif(${QT_PREFIX}Gui_OPENGL_IMPLEMENTATION STREQUAL GLESv2)
+    elseif((${QT_PREFIX}Gui_OPENGL_IMPLEMENTATION STREQUAL GLESv2) OR (QtGui_Enabled_Features MATCHES "opengles"))
         add_definitions(-DOPENGL_ES2)
         list(APPEND OPENGL_LIBS GLESv2)
     elseif(${QT_PREFIX}Gui_OPENGL_IMPLEMENTATION STREQUAL GLES)
